src/mlpack/core/tree/rectangle_tree/rectangle_tree.hpp

```diff
@@ -492,6 +492,10 @@ class RectangleTree
   //! Returns false: this tree type does not have self children.
   static bool HasSelfChildren() { return false; }
 
+  //! Returns false: The first point of this node is not the centroid
+  //! of its bound.
+  static constexpr bool IsFirstPointCentroid() { return false; }
```

Comment:
The idea I had here for `IsFirstPointCentroid()` was to leave it a part of `TreeTraits`, but just to make it a function:

```
template<typename TreeType>
struct TreeTraits
{
  static constexpr bool FirstPointIsCentroid(TreeType*) { return false; }
  ...
}
```

and then for the VP tree we can specialize and make it non-constexpr:

```
template<>
struct TreeTraits<VPTree>
{
  static bool FirstPointIsCentroid(VPTree* v) { ... }
}
```

It changes the syntax of all of the traits, but it keeps them out of the definition of each tree itself, and we are guaranteed that there is a value for these even if the person who wrote the tree did not specify them.  Does that seem reasonable to you?